api/transport/transporttest/reqres.go

```diff
@@ -83,11 +83,6 @@ func (m RequestMatcher) Matches(got interface{}) bool {
 		return false
 	}
 
-	if l.Transport != r.Transport {
```

Comment:
When we bring this back, this would be fine if we added a l.Transport != "" && clause, right? Is the problem that old transports, particularly tests, do not provide the field?

Reply:
Yes.